### DISCUSSION TITLE
Energy Integration: request for self-consumption incentive calculation

### DISCUSSION THREAD

**Discussion by alexdelprete** [home-assistant/frontend#9663] [Other feature requests]
DIscussed this with @bramkragten on discord #devs_energy.

Right now the integration takes into account incentives only for exported energy. In some countries (Italy in my case) you can receive state incentives also for the quota of energy produced and self-consumed.

In the cost/benefit analysis of the integration, it would be nice to be able to configure both incentives.

**Comment by KNXBroker** [comment]
Hello @alexdelprete , are you talking about this ratio? This ratio would be very interesting in my point of view. This ratio gives the incentives to use more consumers during day and not during night. At the moment this ratio can only be guessed by looking at the diagram.
![image](https://user-images.githubusercontent.com/25910703/131263255-2204621a-68b6-4f09-96b8-1a5118ec1eef.png)

**Reply by alexdelprete** [reply]
I'm talking about the financials: right now the energy integration does not allow to consider an incentive for self-consumption. In Italy, with some old state energy agreements, they pay for exported energy AND for self-consumed energy.

> At the moment this ratio can only be guessed by looking at the diagram.

If you look carefully, right under that flow diagram, you'll see this gauge:

![image](https://user-images.githubusercontent.com/7027842/131307276-fbaf721c-0964-445d-9ba0-dcaa0721a7d2.png)

**Comment by wtadler** [comment]
In the U.S. as well, 6 states and Washington, D.C. have [SREC (Solar Renewable Energy Certificate) markets](https://www.solarreviews.com/blog/what-is-an-srec-and-how-can-i-get-the-best-srec-prices) where owners of solar panels sell one SREC for each MWh produced, regardless of whether it is self-consumed or returned to the grid.

Are you imagining visualizing this somehow? I could also see it being built into the Sources section of the energy dashboard, since you effectively get paid (months later) per kWh generated. One issue is you have to guess at the value, since your SRECs are traded on a market. (Though you can also trade through a broker for a guaranteed price.)

![image](https://user-images.githubusercontent.com/6721528/131373275-81a6c67f-e390-49d3-8e6c-81d28d872d71.png)
